### ISSUE TITLE
Visual Studio Build Support...

### ISSUE THREAD

**Issue by datahead8888** [Secretchronicles/TSC#421]
I'd be interested in getting TSC to build in Microsoft Visual Studio.

Aside from finding it useful myself, I think it might help bring some of the Windows programmers on board for the project.  We would probably target the express edition of Visual Studio.

Users would use cmake to generate a Visual Studio .sln solution file.  They could then build the application there.

This task may end up running pretty slowly, but I guess we'll see.

Note that this task is different than #224, which does not necessarily involve Visual Studio.
cc @dlpenguinlover

**Comment by Quintus** [comment]
I leave this to those using MSVC and Windows. I can’t really be of much help here.

Vale,
Quintus

**Comment by datahead8888** [comment]
I started playing around with this.  Currently I am blocked by a PkgConfig error.
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Found rake: C:/Ruby22-x64/bin/rake
Found gperf: C:/Program Files (x86)/GnuWin32/bin/gperf.exe
CMake Error at C:/Program Files (x86)/CMake 2.8/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
  Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
Call Stack (most recent call first):
  C:/Program Files (x86)/CMake 2.8/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:315 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files (x86)/CMake 2.8/share/cmake-2.8/Modules/FindPkgConfig.cmake:102 (find_package_handle_standard_args)
  CMakeLists.txt:166 (find_package)


Configuring incomplete, errors occurred!
See also "C:/TSC/TSC/tsc/build/CMakeFiles/CMakeOutput.log".
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
This seems to be the main site:
http://www.freedesktop.org/wiki/Software/pkg-config/
--If necessary, I can start looking into a source build for pkg-config, but it will slow down this task a bit.

This discussion seems to talk about prebuilt binaries, but it sounds like it requires MinGW.  Is this the best way to go?
http://stackoverflow.com/questions/1710922/how-to-install-pkg-config-in-windows

**Comment by Quintus** [comment]
pkg-config is such a small piece of software that you can just download and compile it yourself. To my knowledge it does not even have other dependencies than a C runtime. Just be sure to set the default search path compilation option for it to something where you actually place your libraries.

Vale,
Quintus

**Comment by Luiji** [comment]
@Quintus pkg-config depends on glib. For this reason many people install pkg-config-lite during bootstrapping functions. http://sourceforge.net/projects/pkgconfiglite

Personally I find it excessive for a tool like pkg-config to require glib when it does such a simple function.

**Comment by datahead8888** [comment]
Yes, I remember reading that it needs glib to compile.  If pkg-config-lite lets me build TSC from source, I will just use that instead.  Thanks, Luiji.

**Comment by Quintus** [comment]
Luiji Maryo <notifications@github.com> writes:

> @Quintus pkg-config depends on glib. For this reason many people
> install pkg-config-lite during bootstrapping
> functions. http://sourceforge.net/projects/pkgconfiglite

Wat O_o. You are right. A tool so small like pkg-config should be
standalone and not have another dependency. I probably misread "glib" as
"glibc" in my package manager’s info, because I could hardly imagine
such a dependency on pkg-config.

Thanks for the link, I need to check it out.

Vale,
Quintus

**Comment by datahead8888** [comment]
pkgconfiglite worked quite well (and came with the binary).  I'll post back if I get stumped again later.  Thanks for the help.

**Comment by datahead8888** [comment]
@georgebr, do you think you might be willing to post the steps you followed if you get TSC compiling in Visual Studio?  Visual Studio has a large user base, so this would help us recruit more programmers.

**Comment by datahead8888** [comment]
@georgerbr, do you think you might be willing to post the steps you followed if you get TSC compiling in Visual Studio?  Visual Studio has a large user base, so this would help us recruit more programmers.

**Comment by georgerbr** [comment]
Sure, it's just that it's a lot of work because of the sheer number of required libraries.

**Comment by datahead8888** [comment]
@georgerbr, Quintus has posted PKGBUILD's for the CEGUI and SDL_mixer.  I understand you should be able to use these to build these dependencies, possibly saving some work.  You might find this helpful to get TSC building in Visual Studio.  See #224.

**Comment by georgerbr** [comment]
Hi @datahead8888,
I had already managed to build all libraries.
The problem now is with the TSC executable. It compiles and links fine, but crashes on startup. I'll try and investigate further.

**Comment by datahead8888** [comment]
Thanks, @georgerbr.  Quintus has been having some difficulty with his Windows build using open source tools.  If you are able to get the Visual Studio build working, let us know and give us a copy of the executable.  This would help us fix #452, which is a critical issue.

**Comment by Quintus** [comment]
> The problem now is with the TSC executable. It compiles and links fine, but crashes on startup. I'll try and investigate further.

You are experienced the same as me then. I managed to compile and link TSC with GCC also, but it crashes on startup (CEGUI exception). Glad to see the behaviour is at least consistent across compilers.

Note for readers: @georgerbr accidentally posted the Visual Studio information [in the wrong ticket](https://github.com/Secretchronicles/TSC/issues/224#issuecomment-147019202). Please continue VS-specific stuff in this ticket, and MinGW-specific stuff in the other ticket.

Valete,
Quintus

**Comment by georgerbr** [comment]
> Thanks, @georgerbr. Quintus has been having some difficulty with his Windows build using open source tools. If you are able to get the Visual Studio build working, let us know and give us a copy of the executable built in debug mode. This would help us debug #452, which is a critical issue.

I worked on this for the last 4 days. and coudn't get it to work.
I used prebuilt components from a number of sources, and it crashed. I recompiled **ALL** libraries bar SDL, with both dynamic and static linking, and it crashed.

So I'll put Visual Studio aside and work on my MinGW dependency package instead.

**Comment by datahead8888** [comment]
@georgerbr - do you think your trouble was related to issue #462?

**Comment by georgerbr** [comment]
> @georgerbr - do you think your trouble was related to issue #462?

No, the crash is in `glibmm` (an `std::bad_alloc`). CEGUI works fine.

**Comment by datahead8888** [comment]
You might post details at http://forum.secretchronicles.de/forums/10 to see if anyone on the team has insight.

The development forum also mirrors to the mailing list.

**Comment by Quintus** [comment]
Mh. A crash in glibmm might indicate a bug in it. If you are after it, you could try to isolate the problem and file a bug report against glibmm. I don’t think the GNOME devs do much testing with MSVC.

Vale,
Quintus

**Comment by Quintus** [comment]
FYI, #556.
